src/components/leaderboard/leaderboard-tab/LeaderboardTab.tsx

```diff
@@ -42,19 +43,32 @@ const LeaderboardItemRank = styled(Paragraph)<ParagraphProps>`
 `;
 
 interface LeaderboardTabProps {
-  tabItems: LeaderboardItem[];
+  tabItems: TabItem[];
   currentPage: number;
   itemsPerPage: number;
   activePanelKey?: number;
 }
 
-export interface LeaderboardItem {
+export interface TabItem {
   rank?: number;
+  id: number;
   name: string; // for now, these are assumed to be unique
   rightSide: React.ReactNode;
   collapseContent?: React.ReactNode;
 }
 
+export const leaderboardItemsToTabItems = (
+  leaderboardItems: LeaderboardItem[],
+): TabItem[] => {
+  return leaderboardItems.map((item) => {
+    return {
+      id: item.id,
+      name: item.name,
+      rightSide: item.blocksCounted,
+    };
+  });
+};
+
```

Comment:
Rather than putting the responsibility on the caller of this component to provide the properly formatted data, I think we should instead have this component take a list of `LeaderboardItem` and internally shape the data as needed. That way, we don't have to store dual states in the parent container.